From 56a90638c1b7c8d7780fc83ca4b775d7d66e70dd Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Fri, 27 Nov 2015 14:44:51 +0800
Subject: [PATCH 0704/5242] MLK-11915-08 ASoC: imx-si476x: init fm_np to avoid
 wild pointer

commit  8f478fcfd3455aa50bcc5cc43cd768ea11241fda from
https://source.codeaurora.org/external/imx/linux-imx.git

init fm_np to avoid wild pointer. Reported by Coverity.

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
(cherry picked from commit dea49f320f33e1f723301a0bfa18573c0f7b5fb1)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-si476x.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/fsl/imx-si476x.c b/sound/soc/fsl/imx-si476x.c
index ea9edba..f40280a 100644
--- a/sound/soc/fsl/imx-si476x.c
+++ b/sound/soc/fsl/imx-si476x.c
@@ -106,7 +106,7 @@ static int imx_si476x_probe(struct platform_device *pdev)
 	struct device_node *ssi_np, *np = pdev->dev.of_node;
 	struct platform_device *ssi_pdev;
 	struct i2c_client *fm_dev;
-	struct device_node *fm_np;
+	struct device_node *fm_np = NULL;
 	int int_port, ext_port, ret;
 
 	ret = of_property_read_u32(np, "mux-int-port", &int_port);
-- 
1.7.9.5

